test/src/test/java/org/corfudb/integration/ServerRestartIT.java

```diff
@@ -810,4 +811,77 @@ public void testCorfuTableMultiIndexReconstruction() throws Exception {
         runtime2.shutdown();
         runtime3.shutdown();
     }
+
+    /**
+     * This test verifies that a stream is rebuilt from a checkpoint, whenever two valid checkpoints exist, but the
+     * latest checkpoint in the log is performed on an earlier snapshot.
+     *
+     * It also verifies that behaviour is kept the same after the node is restarted and both checkpoints
+     * still exist.
+     *
+     * 1. Write 25 entries to stream A.
+     * 2. Start a checkpoint (CP2) at snapshot 15, complete it.
+     * 3. Start a checkpoint (CP1) at snapshot 10, complete it.
+     * 4. Trim on token for CP2 (snapshot = 15).
+     * 5. New runtime instantiate stream A (do a mutation to force to load from checkpoint).
+     * 6. Restart the server
+     * 7. Instantiate map again.
```

Comment:
Please also write the expectation (assertion criteria) from the test.

Reply:
Done.